src/main/kotlin/com/pubnub/api/subscribe/eventengine/effect/HandshakeEffect.kt

```diff
@@ -0,0 +1,42 @@
+package com.pubnub.api.subscribe.eventengine.effect
+
+import com.pubnub.api.PubNubException
+import com.pubnub.api.endpoints.remoteaction.RemoteAction
+import com.pubnub.api.eventengine.EventDeliver
+import com.pubnub.api.eventengine.ManagedEffect
+import com.pubnub.api.subscribe.eventengine.event.Event
+import com.pubnub.api.subscribe.eventengine.event.SubscriptionCursor
+
+class HandshakeEffect(
+    private val handshakeProvider: HandshakeProvider,
+    private val eventDeliver: EventDeliver,
```

Comment:
I would rename eventDeliver to:
interface EventQueue {
    fun add(event: Event)
    fun take() : Event
}